src/mame2003/mame2003.c

```diff
@@ -798,6 +801,101 @@ bool retro_load_game(const struct retro_game_info *game)
    if(!init_game(driverIndex))
     return false;
 
+  #if (HAS_CYCLONE || HAS_DRZ80)
+   int use_cyclone = 1;
+   int use_drz80 = 1;
+   int use_drz80_snd = 1;
+
+	for (i=0;i<NUMGAMES;i++)
+ 	{
+		if (strcmp(drivers[driverIndex]->name,fe_drivers[i].name)==0)
+		{
+			/* ASM cores: 0=None,1=Cyclone,2=DrZ80,3=Cyclone+DrZ80,4=DrZ80(snd),5=Cyclone+DrZ80(snd) */
+         switch (fe_drivers[i].cores)
+         {
+         case 0:
+            use_cyclone = 0;
+				use_drz80_snd = 0;
+				use_drz80 = 0;
+            break;
+         case 1:
+				use_drz80_snd = 0;
+				use_drz80 = 0;
+            break;
+         case 2:
+            use_cyclone = 0;
+            break;
+         case 4:
+            use_cyclone = 0;
+				use_drz80 = 0;
+            break;
+         case 5:
+				use_drz80 = 0;
+            break;
+         default:
+            break;
+         }
+			
+         break;
+		}
+	}
+
+   /* Replace M68000 by CYCLONE */
+#if (HAS_CYCLONE)
+   if (use_cyclone)
+   {
+	   for (i=0;i<MAX_CPU;i++)
+	   {
+		   unsigned int *type=(unsigned int *)&(Machine->drv->cpu[i].cpu_type);
+#ifdef NEOMAME
+		   if (*type==CPU_M68000)
+#else
+			   if (*type==CPU_M68000 || *type==CPU_M68010 )
+#endif
+			   {
+				   *type=CPU_CYCLONE;
+                   log_cb(RETRO_LOG_INFO, LOGPRE "Replaced CPU_CYCLONE\n"); 
+			   }
+        if(!(*type)){
+          break;
+        }
+	   }
+   }
+#endif
+
+#if (HAS_DRZ80)
+	/* Replace Z80 by DRZ80 */
+	if (use_drz80)
+	{
+		for (i=0;i<MAX_CPU;i++)
+		{
+			unsigned int *type=(unsigned int *)&(Machine->drv->cpu[i].cpu_type);
+			if (type==CPU_Z80)
```

Comment:
This seems to be a bug. It should be `if (*type == CPU_Z80)` (gcc does warn about comparison between a pointer and an int). So DRZ80 was never actually enabled because of this.
But after fixing this (as well as on line 887), I lost part of the sounds in games (e.g. `sdfight`), until DRZ80 is turned off.
@frangarcj hope you are still around to take a look.